finality-aleph/src/network/substrate.rs

```diff
@@ -60,163 +83,230 @@ impl ProtocolNaming {
-                        NotificationReceived {
-                            peer,
-                            notification,
-                        } => {
-                            return Some(Messages(
-                                peer,
-                                vec![(Protocol::BlockSync, notification.into())],
-                            ));
-                        },
-                    }
+            match self.sync_stream.next().await {
+                Some(event) => match event {
+                    PeerConnected(remote) => self.peer_connected(remote),
+                    PeerDisconnected(remote) => self.peer_disconnected(remote),
                 },
+                None => return Err(SyncNetworkServiceError::NetworkStreamTerminated),
+            }
+        }
+    }
+}
 
-                Some(event) = self.notifications.authentication.next_event() => {
-                    use SubstrateEvent::*;
-                    match event {
-                        ValidateInboundSubstream {
-                            peer: _,
-                            handshake: _,
-                            result_tx,
-                        } => {
-                            let _ = result_tx.send(ValidationResult::Accept);
-                            continue
-                        },
-                        NotificationStreamOpened {
-                            peer,
-                            ..
-                        } => {
-                            match self.notifications.authentication.message_sink(&peer) {
-                                Some(sink) => return Some(StreamOpened(peer, Protocol::Authentication, sink)),
-                                None => {
-                                    warn!(target: "aleph-network", "Received NotificationStreamOpened from peer {peer:?} in Authentication protocol, but could not create MessageSink.");
-                                    continue;
-                                }
-                            }
-                        },
-                        NotificationStreamClosed {
-                            peer,
-                        } => {
-                            return Some(StreamClosed(peer, Protocol::Authentication));
-                        },
-                        NotificationReceived {
-                            peer,
-                            notification,
-                        } => {
-                            return Some(Messages(
-                                peer,
-                                vec![(Protocol::Authentication, notification.into())],
-                            ));
-                        },
-                    }
-                },
+/// A thin wrapper around sc_network::config::NotificationService that stores a list
+/// of all currently connected peers, and introduces a few convenience methods to
+/// allow broadcasting messages and sending data to random peers.
+pub struct NotificationService {
```

Comment:
Terrible name, we use `Service` to mean something that should be `run` in this codebase. It should probably be called `*Network`, since that is the main purpose of it. Maybe `ProtocolNetwork`?

Reply:
fixed